File: typeScript/1447. 最简分数.ts
function simplifiedFractions(n: number): string[] {
  let arr=[];
  //更相减损
  // let gcd=(a,b)=>{
  //     let [max,min]=a>b?[a,b]:[b,a];
  //     if(max===min){
  //         return max;
  //     }
  //     return gcd(max-min,min)
  // }
  //欧几里得辗转相除O(n*n*logn)
  const gcd=(a,b)=>{
     return a===0?b:gcd(b%a,a)
  }
 for(let i=1;i<n;i++){
     for(let j=i+1;j<=n;j++){
      gcd(i,j)===1?arr.push(`${i}/${j}`):null
     }
 } 
 return arr;
};

File: java/1447. 最简分数.java
class Solution {
    public List<String> simplifiedFractions(int n) {
        //欧几里得 O(n*n*logn)
    List<String> ans = new ArrayList<String>();
        for(int i=1;i<n;i++){
            for(int j=i+1;j<=n;j++){
                if (gcd(i, j) == 1) ans.add(i + "/" + j);
            }
        }
        return ans;
    };
    public int gcd(int a, int b){
            return a==0?b:gcd(b%a,a);
        }
}